src/fidesops/core/config.py

```diff
@@ -59,6 +59,26 @@ class RedisSettings(FidesSettings):
     ENABLED: bool = True
     SSL: bool = False
     SSL_CERT_REQS: Optional[str] = "required"
+    REDIS_CONNECTION_URL: Optional[str] = None
+
+    @validator("REDIS_CONNECTION_URL", pre=True)
+    @classmethod
+    def assemble_redis_connection_url(
+        cls,
+        v: Optional[str],
+        values: Dict[str, str],
+    ) -> str:
+        """Join Redis connection credentials into a connection string"""
+        if isinstance(v, str):
+            # If the whole URL is provided via the config, preference that
+            return v
```

Comment:
Does this need further validation to make sure it is actually a URL? Currently any string would be allowed, but wouldn't actually work.

Reply:
I thought about this — ideally yes it would. I'll need to research the range of connection URIs across vanilla Redis, Elasticache, and whatever other equivalents exist, in order to add anything meaningful though (without resorting to generic [RFC 3986](https://www.rfc-editor.org/rfc/rfc3986) validation). In case we want to pursue that I did add a `USER` to match Redis' own implementation ([even though it's unused](https://stackoverflow.com/a/44353927)).